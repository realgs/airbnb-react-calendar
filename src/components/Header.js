import React from 'react';

const Header = ({ price, currency, rating, reviews }) => {
  const fullStars = Math.floor(rating);
  const halfStars = Math.floor(rating % 1 * 2);
  const blankStars = 5 - fullStars - halfStars;

  const renderFullStars = (number) => {
    return typeof number === 'number' && number > 0 ? [... new Array(number)].map((elem, index) => {
      return <img src="./images/star.png" key={`star${index}`} className="header__rating__star"></img>;
    })
      : '';
  };
  const renderHalfStar = (number) => {
    return typeof number === 'number' && number > 0 ? <img src="./images/halfStar.png" key="halfStar" className="header__rating__halfStar"></img> : '';
  };
  const renderBlankStars = (number) => {
    return typeof number === 'number' && number > 0 ? [... new Array(number)].map((elem, index) => {
      return <img src="./images/blankStar.png" key={`blankStar${index}`} className="header__rating__blankStar"></img>;
    })
      : '';
  };

  return (
    <div className="header">
      <div className="header__price">
        <div className="header__price__value">
          {`${currency}${price}`}
        </div>
        <div className="header__price__note">
          per night
        </div>
      </div>
      <div className="header__rating">
        <div>
          {renderFullStars(fullStars)}
          {renderHalfStar(halfStars)}
          {renderBlankStars(blankStars)}
        </div>
        <div className="header__reviews">{reviews}</div>
      </div>
    </div>
  );
};

export default Header;