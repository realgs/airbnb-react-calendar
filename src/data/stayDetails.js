export const stayDetails = {
  user: 'Matthew.Jings',
  price: 19.9,
  rating : 3.5,
  reviews : 123,
  bonus : 10,
  cleaningFee : 10,
  stayLength : 0,
  minStay: 2,
};