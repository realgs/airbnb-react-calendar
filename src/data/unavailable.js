export const unavailableDates = [
  "2018-11-01",
  "2018-11-02",
  "2018-11-14",
  "2018-11-16",
  "2018-11-20",
  "2018-11-25",
  "2018-11-26",
  "2018-12-01",
  "2018-12-02",
  "2018-12-03",
  "2018-12-15",
  "2018-12-16",
  "2018-12-17",
  "2019-01-05",
  "2019-01-06",
  "2019-01-07",
  "2019-01-08",
  "2019-01-15",
  "2019-01-16",
  "2019-01-17",
  "2019-01-20",
  "2019-01-21",
  "2019-01-22",
  "2019-01-26"
];