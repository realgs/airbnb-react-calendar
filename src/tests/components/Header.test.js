import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../components/Header';
import { stayDetails } from '../../data/stayDetails';

const { price, currency, rating, reviews } = { ...stayDetails };

test('Should match Header snapshot.', () => {
  const wrapper = shallow(<Header
    price={price}
    currency={currency}
    rating={rating}
    reviews={reviews}
  />);
  expect(wrapper).toMatchSnapshot();
});

test('Should render rating as adequate number of stars', () => {
  //const rating = 3.5
  const wrapper = shallow(<Header
    price={price}
    currency={currency}
    rating={rating}
    reviews={reviews}
  />);
  const fullStars = Math.floor(rating);
  const halfStars = Math.floor(rating % 1 * 2);
  const blankStars = 5 - fullStars - halfStars;
  expect(wrapper.find('.header__rating__star').length).toEqual(fullStars);
  expect(wrapper.find('.header__rating__halfStar').length).toEqual(halfStars);
  expect(wrapper.find('.header__rating__blankStar').length).toEqual(blankStars);
  console.log(wrapper.find('.header__rating__blankStar'));
});