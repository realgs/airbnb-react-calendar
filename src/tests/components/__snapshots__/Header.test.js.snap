// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match Header snapshot. 1`] = `
<div
  className="header"
>
  <div
    className="header__price"
  >
    <div
      className="header__price__value"
    >
      undefined19.9
    </div>
    <div
      className="header__price__note"
    >
      per night
    </div>
  </div>
  <div
    className="header__rating"
  >
    <div>
      <img
        className="header__rating__star"
        key="star0"
        src="./images/star.png"
      />
      <img
        className="header__rating__star"
        key="star1"
        src="./images/star.png"
      />
      <img
        className="header__rating__star"
        key="star2"
        src="./images/star.png"
      />
      <img
        className="header__rating__halfStar"
        key="halfStar"
        src="./images/halfStar.png"
      />
      <img
        className="header__rating__blankStar"
        key="blankStar0"
        src="./images/blankStar.png"
      />
    </div>
    <div
      className="header__reviews"
    >
      123
    </div>
  </div>
</div>
`;
